src/main/frontend/state.cljs

```diff
@@ -520,6 +431,242 @@
   []
   (:page-name-order (get-config)))
 
+(defn get-date-formatter
+  []
+  (gp-config/get-date-formatter (get-config)))
+
+(defn shortcuts []
+  (:shortcuts (get-config)))
+
+(defn get-commands
+  []
+  (:commands (get-config)))
+
+(defn get-scheduled-future-days
+  []
+  (let [days (:scheduled/future-days (get-config))]
+    (or (when (int? days) days) 0)))
+
+(defn get-start-of-week
+  []
+  (or (:start-of-week (get-config))
+      (get-in @state [:me :settings :start-of-week])
+      6))
+
+(defn get-ref-open-blocks-level
+  []
+  (or
+    (when-let [value (:ref/default-open-blocks-level (get-config))]
+      (when (integer? value)
+        value))
+    2))
+
+(defn get-linked-references-collapsed-threshold
+  []
+  (or
+    (when-let [value (:ref/linked-references-collapsed-threshold (get-config))]
+      (when (integer? value)
+        value))
+    100))
+
+(defn get-export-bullet-indentation
+  []
+  (case (get (get-config) :export/bullet-indentation :tab)
+    :eight-spaces
+    "        "
+    :four-spaces
+    "    "
+    :two-spaces
+    "  "
+    :tab
+    "\t"))
+
+(defn enable-search-remove-accents?
+  []
+  (:feature/enable-search-remove-accents? (get-config)))
+
+;; State cursor fns for use with rum components
+;; ============================================
+
+;; State that some subs are dependent on. Should they use sub too?
+(declare get-edit-input-id document-mode?)
```

Comment:
It was surprising to see two reactive fns that depended on these non-reactive fns. Should these two have reactive versions?

Reply:
Thanks for pointing out those invalid usage!

For `get-edit-input-id`, it's called in `sub-edit-content` which is only used for the `mock-textarea`, I changed it to subscribe to the passed input `id` instead of using `get-edit-input-id` in 2f8acf439.

For `document-mode`, it's used in `doc-mode-enter-for-new-line?` which is called when there's any keydown or keyup event, so there's no need to be reactive. I fixed it in 133902744.

Reply:
Good to know 👍